src/overlays/actors/ovl_Magic_Fire/z_magic_fire.c

```diff
@@ -47,13 +307,198 @@ static ColliderCylinderInit D_80B8AF50 = {
     },
     { 9, 9, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Magic_Fire/MagicFire_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Magic_Fire/MagicFire_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 0, ICHAIN_STOP),
+};
+
+static u8 sVertexIndices[] = {
+    0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x10, 0x11, 0x12, 0x13, 0x19, 0x1A, 0x1B, 0x20,
+    0x23, 0x24, 0x25, 0x26, 0x27, 0x2D, 0x2E, 0x2F, 0x34, 0x35, 0x36, 0x3B, 0x3C, 0x3D, 0x43, 0x44,
+    0x45, 0x46, 0x47, 0x48, 0x00, 0x01, 0x0B, 0x0C, 0x0E, 0x14, 0x15, 0x17, 0x1C, 0x1E, 0x21, 0x22,
+    0x28, 0x29, 0x2B, 0x30, 0x32, 0x37, 0x39, 0x3E, 0x40, 0x41, 0x49, 0x4A, 0x00, 0x00, 0x00, 0x00,
+};
+
+void MagicFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    MagicFire* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->action = 0;
+    this->screenTintBehaviour = 0;
+    this->actionTimer = 0;
+    this->alphaMultiplier = -3.0f;
+    Actor_SetScale(&this->actor, 0.0f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+    this->actor.update = &MagicFire_UpdateBeforeCast;
+    this->actionTimer = 20;
+    this->actor.room = -1;
+}
+
+void MagicFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    func_800876C8(globalCtx);
+}
+
+void MagicFire_UpdateBeforeCast(Actor* thisx, GlobalContext* globalCtx) {
+    MagicFire* this = THIS;
+    Player* player = PLAYER;
+
+    if ((globalCtx->msgCtx.msgMode == 0xD) || (globalCtx->msgCtx.msgMode == 0x11)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if (this->actionTimer > 0) {
+        this->actionTimer--;
+    } else {
+        this->actor.update = &MagicFire_Update;
+        func_8002F7DC(&player->actor, NA_SE_PL_MAGIC_FIRE);
+    }
+    this->actor.world.pos = player->actor.world.pos;
+}
+
+void MagicFire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    MagicFire* this = THIS;
+    Player* player = PLAYER;
+    s32 pad;
+    if (1) {}
```

Comment:
```suggestion
   
     if (1) {}
```
I think this should be separated from the stack variables.